context/app/static/js/components/workspaces/SelectableTemplateGrid/SelectableTemplateGrid.tsx

```diff
@@ -43,6 +45,22 @@ function SelectableTemplateGrid<FormType extends FormWithTemplates>({
     field.value satisfies FormType[typeof inputName],
   );
 
+  const { field: jobType } = useController({
+    name: 'workspaceJobTypeId' as Path<FormType>,
+    control,
+    rules: { required: true },
+  });
+
+  // If the Python + R job type is selected, select the default R template
+  useEffect(() => {
+    if (jobType.value === R_JOB_TYPE) {
+      setSelectedTemplates([...selectedTemplates, DEFAULT_R_TEMPLATE_KEY]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [jobType.value, setSelectedTemplates]);
```

Comment:
There's undoubtedly a more elegant solution to this - I'd be open to any suggestions!

Reply:
Maybe we could use the `addItem` function from `useSelectItems` to add just the default R template key? That would allow us to skip including `selectedTemplates` as part of the method body. 
https://github.com/hubmapconsortium/portal-ui/blob/47a6036db63188b77cf45322b43abace69887e79/context/app/static/js/hooks/useSelectItems.ts#L49

Reply:
Good idea!